doc/developer-guides/trusty.rst

```diff
@@ -0,0 +1,186 @@
+.. _trusty_tee:
+
+Trusty TEE on ACRN
+##################
+
+Introduction
+************
+
+`Trusty`_ is a set of software components supporting a Trusted Execution Environment (TEE).
+Trusty consists of:
+
+1. An operating system (the Trusty OS) that runs on a processor intended to provide a TEE
+#. Drivers for the Android kernel (Linux) to facilitate communication with applications
+   running under the Trusty OS
+#. A set of libraries for Android/Linux systems software to facilitate communication with
+   trusted applications executed within the Trusty OS using the kernel drivers
+
+LK (`Little Kernel`_) is a tiny operating system suited for small embedded devices, bootloaders,
+and other environments where OS primitives like threads, mutexes, and timers are needed, but
+there's a desire to keep things small and lightweight. LK has been chosen as the Trusty OS kernel.
+
+Trusty Architecture
+*******************
+
+Trusty Architectural diagram::
+
+                      +---------------------------+
+                      |VMn                        |
+                      |        ......             |
+    +------------+  +---------------------------+ |
+    |VM0         |  |VM1                        | |
+    |            |  | +--------+     +--------+ | |
+    |            |  | |        |     |        | | |
+    |    SOS     |  | | Normal |     | Secure | | |
+    |            |  | | World  |     | World  | |-+
+    |            |  | |        |     |        | |
+    |            |  | +--------+     +--------+ |
+    +------------+  +---------------------------+
+    +-------------------------------------------+
+    |             ACRN Hypervisor               |
+    +-------------------------------------------+
+    +-------------------------------------------+
+    |                   HW                      |
+    +-------------------------------------------+
+
+.. note::
+   Trusty OS is running in Secure World in the architecture above.
+
+Trusty specific Hypercalls
+**************************
+
+1. ``HC_INITIALIZE_TRUSTY``
+
+   * This Hypercall is used by the User OS bootloader (``UOS_Loader``) to request ACRN to initialize Trusty
+   * The Trusty memory region range, entry point must be specified
+   * Hypervisor needs to save current vCPU contexts (Normal World)
+#. ``HC_WORLD_SWITCH``
+
+   * Simulate ARM SMC (Secure Monitor Call, or SMC) instruction to do world switch
```

Comment:
Why are we talking about ARM instructions?  Is Trusty ARM-specific and we've ported it to run on Intel processors?

Reply:
No idea :)

@YadongQi @dongyaozu @jren1 , can you enlighten us?